### PR TITLE
libimaginteraction: readline support

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Readline support via the rustyline crate.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@TheNeikos and maybe @neithernut What do you think: Merging this early and fixing it when we start using it (if there are errors), or waiting until some use-case appears (there clearly are, but not implemented yet) and pull in this PR then into the feature-branch of the using feature?

I would simply merge this and go on then.

**Comment by matthiasbeyer** [comment]
I will rebase this now and add the license-header for the new files.

File: imagrc.toml [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+#
+[ui]
+
+#
+# Configuration options for the commandline user interface
+#
+[ui.cli]
+
+# History file path for readline. Will be created by imag if it does not exist.
+readline_history_file = "/tmp/readline.history"
```
Comment (TheNeikos) [inline imagrc.toml]: Hm, I don't think that using `/tmp/` as a default is a good thing, something like `XDG_RUNTIME_DIR` might be better

Comment (matthiasbeyer) [inline imagrc.toml]: We do not (yet?) have ENV-variable expansion in the codebase... so specifying `$XDG_RUNTIME_DIR` won't work.

I'd go for this or something like `~/.imag_readline_history`, though I'm also not sure whether `~` is expanded automatically... and add support for variables later on.

**Comment by matthiasbeyer** [comment]
@TheNeikos Do you want me to edit the config or can I merge this and we wait for env-var expansion support (feel free to file an issue)?

**Comment by TheNeikos** [comment]
The problem with `/tmp/readline.history` is that it is not multi-user, the moment a second user tries to do anything it will error as the file already exists and is not owned by them.

**Comment by matthiasbeyer** [comment]
So we should go for a `~/.imag_readline_history` or something like that. I will change this, but not today! :smile:

**Comment by matthiasbeyer** [comment]
I'm not even sure whether `~` will be expanded correctly... :smiley:

**Comment by matthiasbeyer** [comment]
I will go with the current settings and file an issue for more advanced things (env-variable expansion) as issue.

**Pull request merged**